src/app/genes/gene-similar/gene-similar.component.ts

```diff
@@ -231,33 +246,24 @@ export class GeneSimilarComponent implements OnInit {
     downloadTable() {
         const downloadArray = [];
         // The headers row for the csv file
-        downloadArray[0] = this.cols.map((c) => (c.subfield) ? c.subfield : c.field).join();
+        downloadArray[0] = this.cols.map((c) => c.header).join();
+        // List of columns with values containing commas
+        const escapeFields = ['brain_regions_display_value', 'sm_druggability_display_value',
+            'safety_rating_display_value', 'ab_modality_display_value'];
+
         this.genesInfo.forEach((de: GeneInfo) => {
             downloadArray.push('');
             this.cols.forEach((c, index) => {
-                const curField = (c.subfield) ? c.subfield : c.field;
-                // If we don't have the field in the data file
-                if (de[c.field] === undefined) {
-                    downloadArray[downloadArray.length - 1] += '';
-                } else {
-                    if (c.field === 'druggability') {
-                        downloadArray[downloadArray.length - 1] += de[c.field][0][c.subfield];
-                    } else {
-                        downloadArray[downloadArray.length - 1] += (curField === 'brain_regions' ||
-                            curField === 'pharos_class') ? ('"' + de[curField] + '"') :
-                            de[curField];
-                    }
-                }
-
+                downloadArray[downloadArray.length - 1] +=
+                    // quote the values that contain commas
+                    (escapeFields.includes(c.field)) ?  ('"' + de[c.field] + '"') : de[c.field];
                 downloadArray[downloadArray.length - 1] += ((index) < (this.cols.length - 1)) ?
                     ',' : '';
             });
         });
 
         // Finally export to csv
-        this.dataService.exportToCsv(
-            'genes-similar-' + this.gene.ensembl_gene_id + '.csv', downloadArray
-        );
+        this.dataService.exportToCsv( 'genes-similar-' + this.gene.ensembl_gene_id + '.csv', downloadArray);
```

Comment:
Just wanted to note that it's a little strange to me that we handle building and escaping this CSV client-side, but that's outside of the scope of this PR